packages/@expo/cli/src/api/user/expoSsoLauncher.ts

```diff
@@ -0,0 +1,80 @@
+import openBrowserAsync from 'better-opn';
+import http from 'http';
+import { Socket } from 'node:net';
+import querystring from 'querystring';
+
+import * as Log from '../../log';
+
+export async function getSessionUsingBrowserAuthFlowAsync(options: {
+  expoWebsiteUrl: string;
+  serverPort: number;
+}): Promise<string> {
+  const { expoWebsiteUrl, serverPort } = options;
+  if (!expoWebsiteUrl || !serverPort) {
+    throw new Error('Expo website URL and local server port are required.');
+  }
```

Comment:
```suggestion
```
TypeScript should ensure this. We also use `CommandError` for controlled/known issues when possible.

Reply:
Deleted